plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/webhooks/WebHooksComponent.java

```diff
@@ -0,0 +1,216 @@
+				+ "You can define these triggers using GitHub webhooks or Generic webhooks.");
+		GridDataFactory.fillDefaults()
+			.align(SWT.FILL, SWT.FILL).grab(true, false)
+			.applyTo(webhookExplanation);
+
+//		Label clickExplanation = new Label(this, SWT.NONE);
+//		clickExplanation.setText("Click into a text field to copy its content to the clipboard\n");
+//		GridDataFactory.fillDefaults()
+//			.align(SWT.FILL, SWT.FILL)
+//			.applyTo(clickExplanation);
+
+		Group hooksGroup = new Group(this, SWT.None);
+		hooksGroup.setText("Web Hooks");
+		GridDataFactory.fillDefaults()
+			.align(SWT.FILL, SWT.FILL).grab(true, true)
+			.applyTo(hooksGroup);
+		GridLayoutFactory.fillDefaults()
+			.numColumns(3).margins(10, 10)
+			.applyTo(hooksGroup);
+
+		createHookWidgets(getWebHooks(buildConfig), hooksGroup);
+	}
+
+	private void createHookWidgets(List<IWebhookTrigger> webHooks, Composite parent) {
+		if (webHooks == null
+				|| webHooks.isEmpty()) {
+			createNoHooksMessage(parent);
+		} else {
+			for (IWebhookTrigger webHook : webHooks) {
+				createWebhookWidget(webHook, parent);
+			}
+		}
+	}
+
+	private void createNoHooksMessage(Composite parent) {
+		Label noHooksLabel = new Label(parent, SWT.NONE);
+		noHooksLabel.setText("You have no hooks configured for your build config.");
+	}
+
+	private void createWebhookWidget(IWebhookTrigger webHook, Composite parent) {
+		Link link = new Link(parent, SWT.NONE);
+		link.addSelectionListener(onClickWebhook());
+		String linkLabel = isGitHub(webHook) ? "<a>" + webHook.getType() + "</a>" : webHook.getType();
+		link.setText(linkLabel + " web hook:");
+		GridDataFactory.fillDefaults()
+			.align(SWT.LEFT, SWT.CENTER)
+			.applyTo(link);
+
+		final Text uriText = new Text(parent, SWT.BORDER);
+		uriText.setEditable(false);
+		uriText.setText(webHook.getWebhookURL());
+		uriText.addMouseListener(onClickUriText(uriText));
+		uriText.setToolTipText("Click to copy to the clipboard");
+		GridDataFactory.fillDefaults()
+			.align(SWT.LEFT, SWT.FILL).grab(true, false)
+			.applyTo(uriText);
+		
+		Button copyToClipboard = new Button(parent, SWT.PUSH);
+		copyToClipboard.setImage(OpenShiftImages.COPY_TO_CLIPBOARD_IMG);
+		copyToClipboard.setToolTipText("Copy To Clipboard");
+		copyToClipboard.addSelectionListener(onClickCopyButton(uriText));
+	}
+
+	private SelectionAdapter onClickCopyButton(final Text uriText) {
+		return new SelectionAdapter() {
+
+			@Override
+			public void widgetSelected(SelectionEvent e) {
+				copyToClipBoard(uriText);
+			}
+		};
+	}
+
+	private MouseAdapter onClickUriText(final Text uriText) {
+		return new MouseAdapter() {
+			@Override
+			public void mouseUp(MouseEvent event) {
+				copyToClipBoard(uriText);
+			}
+		};
+	}
+
+	private SelectionAdapter onClickWebhook() {
+		return new SelectionAdapter() {
+			@Override
+			public void widgetSelected(SelectionEvent e) {
+				//TODO open https://github.com/<user>/<repo>/settings/hooks
```

Comment:
This might be more challenging to accomplish when OS supports internal git and other repos.